aries_cloudagent/protocols/issue_credential/v2_0/formats/ld_proof/handler.py

```diff
@@ -353,14 +353,20 @@ def _get_proof_purpose(
                 f"Supported  proof types are: {SUPPORTED_ISSUANCE_PROOF_PURPOSES}"
             )
 
-    async def _prepare_detail(self, detail: LDProofVCDetail) -> LDProofVCDetail:
+    async def _prepare_detail(
+        self, detail: LDProofVCDetail, holder_did: str = None
+    ) -> LDProofVCDetail:
         # Add BBS context if not present yet
         if (
             detail.options.proof_type == BbsBlsSignature2020.signature_type
             and SECURITY_CONTEXT_BBS_URL not in detail.credential.context_urls
         ):
             detail.credential.add_context(SECURITY_CONTEXT_BBS_URL)
 
+        # add holder_did as credentialSubject.id (if provided)
+        if holder_did and holder_did.startswith("did:key"):
```

Comment:
I don't understand why this would be limited to did:key, shouldn't any DID be treated the same here?